nvme-print-stdout.c

```diff
@@ -1427,6 +1427,107 @@ static void stdout_registers_pmrmscu(uint32_t pmrmscu)
 		pmrmscu);
 }
 
+static void stdout_ctrl_register(int offset, uint64_t value64)
+{
+	bool human = stdout_print_ops.flags & VERBOSE;
+	uint32_t value32 = (uint32_t)value64;
+
+	if (human)
+		printf("%s: ", nvme_register_to_string(offset));
+	else
+		printf("register: 0x%02x (%s), value: ", offset, nvme_register_to_string(offset));
```

Comment:
The output example is below also the human-readable option is for readable format output but not for verbose output.
```sh
tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme get-reg /dev/nvme0 -c
register: 0x00 (Controller Capabilities), value: 3028033fff
tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme get-reg /dev/nvme0 -c -H
Controller Capabilities: 3028033fff
        Controller Ready With Media Support (CRWMS): Not Supported
        Controller Ready Independent of Media Support (CRIMS): Not Supported
        NVM Subsystem Shutdown Supported   (NSSS): Not Supported
        Controller Memory Buffer Supported (CMBS): The Controller Memory Buffer is Not Supported
        Persistent Memory Region Supported (PMRS): The Persistent Memory Region is Not Supported
        Memory Page Size Maximum         (MPSMAX): 4096 bytes
        Memory Page Size Minimum         (MPSMIN): 4096 bytes
        Controller Power Scope              (CPS): Not Reported
        Boot Partition Support              (BPS): No
        Command Sets Supported              (CSS): NVM command set is Supported
                                                   One or more I/O Command Sets are Not Supported
                                                   I/O Command Set is Supported
        NVM Subsystem Reset Supported     (NSSRS): Yes
        Doorbell Stride                   (DSTRD): 4 bytes
        Timeout                              (TO): 20000 ms
        Arbitration Mechanism Supported     (AMS): Weighted Round Robin with Urgent Priority Class is Not supported
        Contiguous Queues Required          (CQR): Yes
        Maximum Queue Entries Supported    (MQES): 16384

tokunori@tokunori-desktop:~/nvme-cli$ sudo .build/nvme get-reg /dev/nvme0 -c -h
Usage: nvme get-reg <device> [OPTIONS]

Reads and shows the defined NVMe controller register.
Register offset must be one of:
CAP=0x0, VS=0x8, INTMS=0xc, INTMC=0x10, CC=0x14, CSTS=0x1c,
NSSR=0x20, AQA=0x24, ASQ=0x28, ACQ=0x30, CMBLOC=0x38,
CMBSZ=0x3c, BPINFO=0x40, BPRSEL=0x44, BPMBL=0x48, CMBMSC=0x50,
CMBSTS=0x58, CRTO=0x68, PMRCAP=0xe00, PMRCTL=0xe04,
PMRSTS=0xe08, PMREBS=0xe0c, PMRSWTP=0xe10, PMRMSCL=0xe14, PMRMSCU=0xe18

Options:
  [  --verbose, -v ]                    --- Increase output verbosity
  [  --output-format=<FMT>, -o <FMT> ]  --- Output format: normal|json|binary
  [  --offset=<NUM>, -O <NUM> ]         --- offset of the requested register
  [  --human-readable, -H ]             --- show register in readable format
```

Reply:
> Also why use `%02x`? There are already some offsets that take 3 hex digits.

Fixed by the commit 24f79e35.

Reply:
But in this specific line (`register: 0x00 (Controller Capabilities), value: 3028033fff` vs. `Controller Capabilities: 3028033fff`), the verbose output doesn't have the register number. It seems like "verbose" should at least contain all the information that "not verbose" does.

Reply:
To make sure let me reconfirm as the "verbose" mentioned is not actually set by the command verbose option but set by the human-readable so not necessary to contain the register number for the option output.
```c
static int get_register(int argc, char **argv, struct command *cmd, struct plugin *plugin)
{
...
	bool human = stdout_print_ops.flags & VERBOSE;
...
static void stdout_ctrl_register(int offset, uint64_t value64)
{
	if (cfg.human_readable)
		flags |= VERBOSE;
```

Reply:
I think `stdout_ctrl_register_common` still has this issue. 

```diff
+static void stdout_ctrl_register_common(int offset, uint64_t value, bool fabrics)
+{
+       bool human = !!(stdout_print_ops.flags & VERBOSE);
+       const char *name = nvme_register_to_string(offset);
+       const char *type = fabrics ? "property" : "register";
+
+       if (human) {
+               printf("%s: %#"PRIx64"\n", name, value);
+               stdout_ctrl_register_human(offset, value, true);
+               return;
+       }
+
+       printf("%s: %#04x (%s), value: %#"PRIx64"\n", type, offset,
+              name, value);
+}
```

While `stdout_ctrl_register_support` handle the verbose flag differently, that is it prints the normal output and then the verbose. This is what I would expect from the previous output as well

```diff
+static void stdout_ctrl_register_support(void *bar, bool fabrics, int offset, bool human,
+                                        bool support)
+{
+       uint64_t value = nvme_is_64bit_reg(offset) ? mmio_read64(bar + offset) :
+           mmio_read32(bar + offset);
+
+       if (fabrics && value == -1)
+               return;
+
+       printf("%-8s: ", nvme_register_symbol_to_string(offset));
+
+       printf("%#"PRIx64"\n", value);
+
+       if (human)
+               stdout_ctrl_register_human(offset, value, support);
+}
```

Reply:
No `stdout_ctrl_register_common()` returns after the verbose output and before it outputs also the normal output. So the order itself same with the `stdout_ctrl_register_support()`.
Note: The normal outputs below are different since depended on the current `show-regs` command outputs.
```c
stdout_ctrl_register_support
static void stdout_ctrl_register_common(int offset, uint64_t value, bool fabrics)
{
...
	if (human) {
		printf("%s: %#"PRIx64"\n", name, value);
...
	printf("%s: %#04x (%s), value: %#"PRIx64"\n", type, offset,
	       name, value);
}
```